superduper/base/datalayer.py

```diff
@@ -402,6 +404,24 @@ def load(
             )
             uuid = info['uuid']
             info.update(overrides or {})
+        else:
+            if (component, identifier) in self._component_cache:
+
+                assert isinstance(identifier, str)
+                if self._component_cache[
+                    (component, identifier)
+                ].uuid == self.metadata.get_latest_uuid(
+                    component=component,
+                    identifier=identifier,
+                ):
+                    logging.info(f'Found {component, identifier} in cache...')
+                    return self._component_cache[(component, identifier)]
+                else:
+                    logging.info(
+                        f'Found {component, identifier} '
+                        'in cache but UUID does not match...'
+                    )
+                    del self._component_cache[(component, identifier)]
```

Comment:
There should be a parameter to control whether to force a reload and bypass the cache.